libdnf/base/goal.cpp

```diff
@@ -828,6 +831,13 @@ base::Transaction Goal::resolve(bool allow_erasing) {
         p_impl->rpm_goal.set_installonly_limit(cfg_main.installonly_limit().get_value());
     }
     ret |= p_impl->rpm_goal.resolve();
+    if ((ret & libdnf::GoalProblem::REMOVAL_OF_PROTECTED) == libdnf::GoalProblem::REMOVAL_OF_PROTECTED) {
```

Comment:
`(ret & libdnf::GoalProblem::REMOVAL_OF_PROTECTED)` should be enough.

Reply:
Not really, because libdnf::GoalProblem is a enum class where values cannot be converted to bool - it is not `enum` represented by `int`.

Reply:
Oh I see. Sorry for the confusion.